src/placeos-driver/queue.cr

```diff
@@ -126,18 +121,29 @@ class PlaceOS::Driver::Queue
       else
         # re-queue the current task
         priority = task.priority + @retry_bonus
-        queue_task(priority, task)
+        @mutex.synchronize { queue_task(priority, task) }
```

Comment:
I think having the synchronisation hidden and closer data access itself is better
So rather than lock here, lock inside `queue_task`